File: exercicios/arrays/arrays_8.js
let buscaBinaria = [];
let escolha = 400;
let encontrado = false;

for (let i = 0; i <= 1000; i++) {
    buscaBinaria[i] = i + 1;
}

let menor = 0;
let maior = buscaBinaria.length - 1;

while (menor <= maior) {
    let meio = Math.floor((menor + maior) / 2);

    if (buscaBinaria[meio] === escolha) {
        console.log("Achei no índice:", meio);
        encontrado = true;
        break;
    } else if (buscaBinaria[meio] < escolha) {
        menor = meio + 1;
    } else {
        maior = meio - 1;
    }
}

if (!encontrado) {
    console.log("Não encontrado");
}

File: exercicios/arrays/arrays_8.py
buscaBinaria = []

for i in range(1, 1001):
    buscaBinaria.append(i)

escolha = int(input("Escolha um número: "))
menor = 0
maior = len(buscaBinaria) - 1
encontrado = False

while menor <= maior:
    meio = (maior + menor) // 2

    if buscaBinaria[meio] == escolha:
        print("Achei!")
        encontrado = True
        break
    elif buscaBinaria[meio] < escolha:
        menor = meio + 1
    else:
        maior = meio - 1

if not encontrado:
    print("Não encontrado.")
